Stripe/STPSource.m

```diff
@@ -289,15 +287,38 @@ - (UIImage *)templateImage {
 }
 
 - (NSString *)label {
-    if (self.type == STPSourceTypeCard
-        && self.cardDetails != nil) {
-        NSString *brand = [STPCard stringFromBrand:self.cardDetails.brand];
-        return [NSString stringWithFormat:@"%@ %@", brand, self.cardDetails.last4];;
-    }
-    else {
-        return [STPCard stringFromBrand:STPCardBrandUnknown];
+    switch (self.type) {
+        case STPSourceTypeBancontact:
+            return @"Bancontact";
+        case STPSourceTypeCard:
+            if (self.cardDetails != nil) {
+                NSString *brand = [STPCard stringFromBrand:self.cardDetails.brand];
+                return [NSString stringWithFormat:@"%@ %@", brand, self.cardDetails.last4];
+            }
+            else {
+                return [STPCard stringFromBrand:STPCardBrandUnknown];
+            }
+        case STPSourceTypeGiropay:
+            return @"Giropay";
+        case STPSourceTypeIDEAL:
+            return @"iDEAL";
+        case STPSourceTypeSEPADebit:
+            return @"SEPA Direct Debit";
+        case STPSourceTypeSofort:
+            return @"SOFORT";
+        case STPSourceTypeThreeDSecure:
+            return @"3D Secure";
+        case STPSourceTypeAlipay:
+            return @"Alipay";
+        case STPSourceTypeP24:
+            return @"P24";
+        case STPSourceTypeEPS:
+            return @"EPS";
+        case STPSourceTypeMultibanco:
+            return @"Multibanco";
```

Comment:
Do any of these need to be localizable strings?

My first thought was no, because they're brand/product names. On the other hand, I suspect it'd be weird to see "Alipay" instead of "支付宝" (according to wikipedia) in an app localized to Chinese.

Reply:
Good point, we should probably wrap them in NSLocalizedString calls in case users want to localize it themselves in a particular way.